src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/ExternalMovementsController.kt

```diff
@@ -45,20 +50,22 @@ class ExternalMovementsController(val externalMovementService: ExternalMovementS
     @RequestParam(defaultValue = "date") sortColumn: String,
     @RequestParam(defaultValue = "false") sortedAsc: Boolean,
     @RequestParam direction: String?,
+    @RequestParam startDate: LocalDate?,
+    @RequestParam endDate: LocalDate?,
   ): List<ExternalMovement> {
     return externalMovementService.list(
       selectedPage = selectedPage,
       pageSize = pageSize,
       sortColumn = sortColumn,
       sortedAsc = sortedAsc,
-      filters = createFilterMap(direction),
+      filters = createFilterMap(direction, startDate, endDate),
     )
   }
 
-  private fun createFilterMap(direction: String?): Map<ExternalMovementFilter, String> =
+  private fun createFilterMap(direction: String?, startDate: LocalDate?, endDate: LocalDate?): Map<ExternalMovementFilter, Any> =
     buildMap {
-      if (!direction.isNullOrBlank()) {
-        put(DIRECTION, direction)
-      }
+      direction?.trim()?.ifEmpty { null }?.let { put(DIRECTION, it) }
```

Comment:
Minor - I think if we re-phrase this slightly it might be slightly easier to read e.g. 

```direction?.trim()?.let { if (it.isNotEmpty()) put(DIRECTION, it) }```

Reply:
Yes it reads better. Thanks, I have now changed that.